sdk/python/feast/feature_store.py

```diff
@@ -1012,6 +1012,18 @@ def teardown(self):
         self._get_provider().teardown_infra(self.project, tables, entities)
         self._registry.teardown()
 
+    @log_exceptions_and_usage
+    def teardown_project(self):
```

Comment:
this _seems_ wrong since the registry file or sql registry itself still holds the references to the project. Probably better is to have each registry store also support tearing down references specific to a project?

Reply:
💯 it's not as simple as first expected since the registries don't have parity in their behavior. I changed the PR to draft to chew on it a big longer and think through how each type of registry would solve this pattern. Thanks for the feedback @adchia 🙌🏻